mlrun/api/db/sqldb/db.py

```diff
@@ -949,15 +956,15 @@ def _update_feature_set_entities(
             feature_set.entities.append(entity)
 
     def _update_feature_set_spec(
-        self, feature_set: FeatureSet, new_feature_set_dict: dict
+        self, feature_set: FeatureSet, new_feature_set_dict: dict, replace=True
     ):
         feature_set_spec = new_feature_set_dict.get("spec")
         if feature_set_spec:
             features = feature_set_spec.pop("features", [])
             entities = feature_set_spec.pop("entities", [])
 
-        self._update_feature_set_features(feature_set, features)
-        self._update_feature_set_entities(feature_set, entities)
+        self._update_feature_set_features(feature_set, features, replace)
+        self._update_feature_set_entities(feature_set, entities, replace)
```

Comment:
I think these 2 lines should be indented inside the if no ?

Reply:
No. They should be like this - if either `features` or `entities` in the modified object are missing, `None` will be passed to the `_update_feature_set_*` method and they will be removed from the DB object. That's expected.